tests/azure_iot_provisioning_client_ut.c

```diff
@@ -0,0 +1,704 @@
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTProvisioningClient_SymmetricKeySet_Failure( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+
+    ( void ) ppvState;
+
+    /* Fail AzureIoTProvisioningClient_SetSymmetricKey when null symmetric key is passed */
+    assert_int_not_equal( AzureIoTProvisioningClient_SetSymmetricKey( &xTestProvisioningClient,
+                                                                      NULL, 0,
+                                                                      prvHmacFunction ),
+                          eAzureIoTProvisioningSuccess );
+
+    /* Fail AzureIoTProvisioningClient_SetSymmetricKey when null hashing function is passed */
+    assert_int_not_equal( AzureIoTProvisioningClient_SetSymmetricKey( &xTestProvisioningClient,
+                                                                      ucSymmetricKey, sizeof( ucSymmetricKey ) - 1,
+                                                                      NULL ),
+                          eAzureIoTProvisioningSuccess );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTProvisioningClient_SymmetricKeySet_Success( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+
+    ( void ) ppvState;
+
+    assert_int_equal( AzureIoTProvisioningClient_SetSymmetricKey( &xTestProvisioningClient,
+                                                                  ucSymmetricKey, sizeof( ucSymmetricKey ) - 1,
+                                                                  prvHmacFunction ),
+                      eAzureIoTProvisioningSuccess );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTProvisioningClient_Register_ConnectFailure( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+    AzureIoTProvisioningClientResult_t xResult;
+
+    ( void ) ppvState;
+
+    prvSetupTestProvisioningClient( &xTestProvisioningClient );
+
+    /* Connect */
+    will_return( prvHmacFunction, 0 );
+    will_return( AzureIoTMQTT_Connect, eAzureIoTMQTTServerRefused );
+    will_return( AzureIoTMQTT_ProcessLoop, eAzureIoTMQTTSuccess );
+    xResult = AzureIoTProvisioningClient_Register( &xTestProvisioningClient, 0 );
+    assert_int_not_equal( xResult, eAzureIoTProvisioningPending );
+    assert_int_not_equal( xResult, eAzureIoTProvisioningSuccess );
+
+    AzureIoTProvisioningClient_Deinit( &xTestProvisioningClient );
+}
+/*-----------------------------------------------------------*/
+
+static void testAzureIoTProvisioningClient_Register_SubscribeFailure( void ** ppvState )
+{
+    AzureIoTProvisioningClient_t xTestProvisioningClient;
+    AzureIoTProvisioningClientResult_t xResult;
+
+    ( void ) ppvState;
+
+    prvSetupTestProvisioningClient( &xTestProvisioningClient );
+
+    /* Connect */
+    will_return( prvHmacFunction, 0 );
```

Comment:
Nit: Consider making these 4 lines of doing the will_return to Register call into a helper function, since it's called a bunch and if anything changes in the implementation it'll be easier to change 1 helper that bunches of copy/paste.

Reply:
sure.

---
In reply to: [607957371](https://github.com/Azure/azure-iot-middleware-freertos/pull/62#discussion_r607957371) [](ancestors = 607957371)